src/friend_foe.py

```diff
@@ -0,0 +1,10 @@
+"""This module selects friends with 4 letters in their name."""
+
+
+def friend(x):
+    """Build list of friends with 4 letters."""
+    friend_list = []
+    for i, c in enumerate(x):
```

Comment:
`enumerate` is great, but this is a mis-use of it. In this case, you don't really need the index, so it would be best to just iterate over the values:
`for friend in x:`

Even more so, think about how you could use a single-line list comprehension instead.